File: sample/java/src/test/java/com/vxlisp/vx/data/FileTest.java

package com.vxlisp.vx.data;

import java.util.Arrays;
import java.util.ArrayList;
import java.util.List;
import com.vxlisp.vx.*;


public final class FileTest {

  static Test.Type_testcase f_boolean_exists_from_file(final Core.Type_context context) {
    Test.Type_testcase output = Test.t_testcase.vx_new(
      ":passfail", false,
      ":testpkg", "vx/data/file",
      ":casename", "boolean-exists<-file",
      ":describelist",
      Test.t_testdescribelist.vx_new(
        Test.t_testdescribe.vx_new(
          ":describename", "(test-true\n (boolean-exists<-file\n  (file\n   :name \"boolean_exists_from_file.txt\"\n   :path \"src/test/resources/vx\")))",
          ":testresult",
            Test.f_test_true(
              File.f_boolean_exists_from_file(
                Core.f_new(
                  File.t_file,
                  Core.t_anylist.vx_new(
                    Core.t_string.vx_new_from_string(":name"),
                    Core.t_string.vx_new_from_string("boolean_exists_from_file.txt"),
                    Core.t_string.vx_new_from_string(":path"),
                    Core.t_string.vx_new_from_string("src/test/resources/vx")
                  )
                )
              ),
              context
            )
        )
      )
    );
    return output;
  }

  static Test.Type_testcase f_boolean_write_from_file_string(final Core.Type_context context) {
    Test.Type_testcase output = Test.t_testcase.vx_new(
      ":passfail", false,
      ":testpkg", "vx/data/file",
      ":casename", "boolean-write<-file-string",
      ":describelist",
      Test.t_testdescribelist.vx_new(
        Test.t_testdescribe.vx_new(
          ":describename", "(test-true\n (boolean-write<-file-string\n  (file\n   :name \"boolean_write_from_file_string.txt\"\n   :path \"src/test/resources/vx\")\n  \"writetext\"))",
          ":testresult",
            Test.f_test_true(
              File.f_boolean_write_from_file_string(
                Core.f_new(
                  File.t_file,
                  Core.t_anylist.vx_new(
                    Core.t_string.vx_new_from_string(":name"),
                    Core.t_string.vx_new_from_string("boolean_write_from_file_string.txt"),
                    Core.t_string.vx_new_from_string(":path"),
                    Core.t_string.vx_new_from_string("src/test/resources/vx")
                  )
                ),
                Core.t_string.vx_new_from_string("writetext"),
                context
              ),
              context
            )
        )
      )
    );
    return output;
  }

  static Test.Type_testcase f_file_read_from_file(final Core.Type_context context) {
    Test.Type_testcase output = Test.t_testcase.vx_new(
      ":passfail", false,
      ":testpkg", "vx/data/file",
      ":casename", "file-read<-file",
      ":describelist",
      Test.t_testdescribelist.vx_new(
        Test.t_testdescribe.vx_new(
          ":describename", "(test\n (file\n  :name \"string_read_from_file.txt\"\n  :path \"src/test/resources/vx\"\n  :text \"testdata\")\n (file-read<-file\n  (file\n   :name \"string_read_from_file.txt\"\n   :path \"src/test/resources/vx\")))",
          ":testresult",
            Test.f_test(
              Core.f_new(
                File.t_file,
                Core.t_anylist.vx_new(
                  Core.t_string.vx_new_from_string(":name"),
                  Core.t_string.vx_new_from_string("string_read_from_file.txt"),
                  Core.t_string.vx_new_from_string(":path"),
                  Core.t_string.vx_new_from_string("src/test/resources/vx"),
                  Core.t_string.vx_new_from_string(":text"),
                  Core.t_string.vx_new_from_string("testdata")
                )
              ),
              File.f_file_read_from_file(
                Core.f_new(
                  File.t_file,
                  Core.t_anylist.vx_new(
                    Core.t_string.vx_new_from_string(":name"),
                    Core.t_string.vx_new_from_string("string_read_from_file.txt"),
                    Core.t_string.vx_new_from_string(":path"),
                    Core.t_string.vx_new_from_string("src/test/resources/vx")
                  )
                ),
                context
              ),
              context
            )
        )
      )
    );
    return output;
  }

  static Test.Type_testcase f_pathcurrent_from_os(final Core.Type_context context) {
    Test.Type_testcase output = Test.t_testcase.vx_new(
      ":passfail", false,
      ":testpkg", "vx/data/file",
      ":casename", "pathcurrent<-os",
      ":describelist",
      Test.t_testdescribelist.vx_new(
        Test.t_testdescribe.vx_new(
          ":describename", "(test-true\n (is-endswith (pathcurrent<-os) \"\\java\"))",
          ":testresult",
            Test.f_test_true(
              Core.f_is_endswith(
                File.f_pathcurrent_from_os(),
                Core.t_string.vx_new_from_string("\\java")
              ),
              context
            )
        )
      )
    );
    return output;
  }

  static Test.Type_testcase f_string_read_from_file(final Core.Type_context context) {
    Test.Type_testcase output = Test.t_testcase.vx_new(
      ":passfail", false,
      ":testpkg", "vx/data/file",
      ":casename", "string-read<-file",
      ":describelist",
      Test.t_testdescribelist.vx_new(
        Test.t_testdescribe.vx_new(
          ":describename", "(test\n \"testdata\"\n (string-read<-file\n  (file\n   :name \"string_read_from_file.txt\"\n   :path \"src/test/resources/vx\")))",
          ":testresult",
            Test.f_test(
              Core.t_string.vx_new_from_string("testdata"),
              File.f_string_read_from_file(
                Core.f_new(
                  File.t_file,
                  Core.t_anylist.vx_new(
                    Core.t_string.vx_new_from_string(":name"),
                    Core.t_string.vx_new_from_string("string_read_from_file.txt"),
                    Core.t_string.vx_new_from_string(":path"),
                    Core.t_string.vx_new_from_string("src/test/resources/vx")
                  )
                ),
                context
              ),
              context
            )
        )
      )
    );
    return output;
  }

  public static Test.Type_testcaselist test_cases(final Core.Type_context context) {
    List<Core.Type_any> arraylisttestcase = new ArrayList<>(Arrays.asList(
      FileTest.f_boolean_exists_from_file(context),
      FileTest.f_boolean_write_from_file_string(context),
      FileTest.f_file_read_from_file(context),
      FileTest.f_pathcurrent_from_os(context),
      FileTest.f_string_read_from_file(context)
    ));
    Test.Type_testcaselist output = Test.t_testcaselist.vx_new(arraylisttestcase);
    return output;
  }

  public static Test.Type_testcoveragesummary test_coveragesummary() {
    return Test.t_testcoveragesummary.vx_new(
      ":testpkg",   "vx/data/file", 
      ":constnums", Test.t_testcoveragenums.vx_new(":pct", 100, ":tests", 0, ":total", 0), 
      ":docnums", Test.t_testcoveragenums.vx_new(":pct", 90, ":tests", 10, ":total", 11), 
      ":funcnums", Test.t_testcoveragenums.vx_new(":pct", 55, ":tests", 5, ":total", 9), 
      ":ospacenums", Test.t_testcoveragenums.vx_new(":pct", 0, ":tests", 0, ":total", 9), 
      ":otimenums", Test.t_testcoveragenums.vx_new(":pct", 0, ":tests", 0, ":total", 9), 
      ":totalnums", Test.t_testcoveragenums.vx_new(":pct", 45, ":tests", 5, ":total", 11), 
      ":typenums", Test.t_testcoveragenums.vx_new(":pct", 0, ":tests", 0, ":total", 2)
    );
  }

  public static Test.Type_testcoveragedetail test_coveragedetail() {
    return Test.t_testcoveragedetail.vx_new(
      ":testpkg", "vx/data/file",
      ":typemap", Core.t_intmap.vx_new(
        ":file", 0,
        ":fileformat", 0
      ),
      ":constmap", Core.t_intmap.vx_new(
  
      ),
      ":funcmap", Core.t_intmap.vx_new(
        ":boolean-exists<-file", 1,
        ":boolean-write<-file-any", 0,
        ":boolean-write<-file-string", 1,
        ":file-read<-file", 1,
        ":name<-file", 0,
        ":path<-file", 0,
        ":pathcurrent<-os", 1,
        ":pathfull<-file", 0,
        ":string-read<-file", 1
      )
    );
  }

  public static Test.Type_testpackage test_package(final Core.Type_context context) {
    Test.Type_testcaselist testcaselist = test_cases(context);
    Test.Type_testpackage output = Test.t_testpackage.vx_new(
      ":testpkg", "vx/data/file", 
      ":caselist", testcaselist,
      ":coveragesummary", test_coveragesummary(),
      ":coveragedetail", test_coveragedetail()
    );
    return output;
  }

}

File: core/cpp/test/vx/data/file_test.cpp
#include <vector>
#include "../../../src/vx/data/file.hpp"
#include "../../../src/vx/core.hpp"
#include "../../../src/vx/test.hpp"
#include "../../../src/vx/data/textblock.hpp"

namespace vx_data_file_test {

  vx_test::Type_testcase* f_boolean_exists_from_file(vx_core::Type_context* context) {
    vx_test::Type_testcase* output = vx_test::t_testcase->vx_new(
      vx_core::t_string->vx_new_from_string(":passfail"), vx_core::c_false,
      vx_core::t_string->vx_new_from_string(":testpkg"), vx_core::t_string->vx_new_from_string("vx/data/file"),
      vx_core::t_string->vx_new_from_string(":casename"), vx_core::t_string->vx_new_from_string("boolean-exists<-file"),
      vx_core::t_string->vx_new_from_string(":describelist"),
      vx_test::t_testdescribelist->vx_new(
        vx_test::t_testdescribe->vx_new(
          vx_core::t_string->vx_new_from_string(":describename"), vx_core::t_string->vx_new_from_string("(test-true\n (boolean-exists<-file\n  (file\n   :name \"boolean_exists_from_file.txt\"\n   :path \"src/test/resources/vx\")))"),
          vx_core::t_string->vx_new_from_string(":testresult"),
            vx_test::f_test_true(
              vx_data_file::f_boolean_exists_from_file(
                vx_core::f_new(
                  vx_data_file::t_file,
                  vx_core::t_anylist->vx_new(
                    vx_core::t_string->vx_new_from_string(":name"),
                    vx_core::t_string->vx_new_from_string("boolean_exists_from_file.txt"),
                    vx_core::t_string->vx_new_from_string(":path"),
                    vx_core::t_string->vx_new_from_string("src/test/resources/vx")
                  )
                )
              ),
              context
            )
        )
      )
    );
    return output;
  }

  vx_test::Type_testcase* f_boolean_write_from_file_string(vx_core::Type_context* context) {
    vx_test::Type_testcase* output = vx_test::t_testcase->vx_new(
      vx_core::t_string->vx_new_from_string(":passfail"), vx_core::c_false,
      vx_core::t_string->vx_new_from_string(":testpkg"), vx_core::t_string->vx_new_from_string("vx/data/file"),
      vx_core::t_string->vx_new_from_string(":casename"), vx_core::t_string->vx_new_from_string("boolean-write<-file-string"),
      vx_core::t_string->vx_new_from_string(":describelist"),
      vx_test::t_testdescribelist->vx_new(
        vx_test::t_testdescribe->vx_new(
          vx_core::t_string->vx_new_from_string(":describename"), vx_core::t_string->vx_new_from_string("(test-true\n (boolean-write<-file-string\n  (file\n   :name \"boolean_write_from_file_string.txt\"\n   :path \"src/test/resources/vx\")\n  \"writetext\"))"),
          vx_core::t_string->vx_new_from_string(":testresult"),
            vx_test::f_test_true(
              vx_data_file::f_boolean_write_from_file_string(
                vx_core::f_new(
                  vx_data_file::t_file,
                  vx_core::t_anylist->vx_new(
                    vx_core::t_string->vx_new_from_string(":name"),
                    vx_core::t_string->vx_new_from_string("boolean_write_from_file_string.txt"),
                    vx_core::t_string->vx_new_from_string(":path"),
                    vx_core::t_string->vx_new_from_string("src/test/resources/vx")
                  )
                ),
                vx_core::t_string->vx_new_from_string("writetext"),
                context
              ),
              context
            )
        )
      )
    );
    return output;
  }

  vx_test::Type_testcase* f_file_read_from_file(vx_core::Type_context* context) {
    vx_test::Type_testcase* output = vx_test::t_testcase->vx_new(
      vx_core::t_string->vx_new_from_string(":passfail"), vx_core::c_false,
      vx_core::t_string->vx_new_from_string(":testpkg"), vx_core::t_string->vx_new_from_string("vx/data/file"),
      vx_core::t_string->vx_new_from_string(":casename"), vx_core::t_string->vx_new_from_string("file-read<-file"),
      vx_core::t_string->vx_new_from_string(":describelist"),
      vx_test::t_testdescribelist->vx_new(
        vx_test::t_testdescribe->vx_new(
          vx_core::t_string->vx_new_from_string(":describename"), vx_core::t_string->vx_new_from_string("(test\n (file\n  :name \"string_read_from_file.txt\"\n  :path \"src/test/resources/vx\"\n  :text \"testdata\")\n (file-read<-file\n  (file\n   :name \"string_read_from_file.txt\"\n   :path \"src/test/resources/vx\")))"),
          vx_core::t_string->vx_new_from_string(":testresult"),
            vx_test::f_test(
              vx_core::f_new(
                vx_data_file::t_file,
                vx_core::t_anylist->vx_new(
                  vx_core::t_string->vx_new_from_string(":name"),
                  vx_core::t_string->vx_new_from_string("string_read_from_file.txt"),
                  vx_core::t_string->vx_new_from_string(":path"),
                  vx_core::t_string->vx_new_from_string("src/test/resources/vx"),
                  vx_core::t_string->vx_new_from_string(":text"),
                  vx_core::t_string->vx_new_from_string("testdata")
                )
              ),
              vx_data_file::f_file_read_from_file(
                vx_core::f_new(
                  vx_data_file::t_file,
                  vx_core::t_anylist->vx_new(
                    vx_core::t_string->vx_new_from_string(":name"),
                    vx_core::t_string->vx_new_from_string("string_read_from_file.txt"),
                    vx_core::t_string->vx_new_from_string(":path"),
                    vx_core::t_string->vx_new_from_string("src/test/resources/vx")
                  )
                ),
                context
              ),
              context
            )
        )
      )
    );
    return output;
  }

  vx_test::Type_testcase* f_pathcurrent_from_os(vx_core::Type_context* context) {
    vx_test::Type_testcase* output = vx_test::t_testcase->vx_new(
      vx_core::t_string->vx_new_from_string(":passfail"), vx_core::c_false,
      vx_core::t_string->vx_new_from_string(":testpkg"), vx_core::t_string->vx_new_from_string("vx/data/file"),
      vx_core::t_string->vx_new_from_string(":casename"), vx_core::t_string->vx_new_from_string("pathcurrent<-os"),
      vx_core::t_string->vx_new_from_string(":describelist"),
      vx_test::t_testdescribelist->vx_new(
        vx_test::t_testdescribe->vx_new(
          vx_core::t_string->vx_new_from_string(":describename"), vx_core::t_string->vx_new_from_string("(test-true\n (is-endswith (pathcurrent<-os) \"\\java\"))"),
          vx_core::t_string->vx_new_from_string(":testresult"),
            vx_test::f_test_true(
              vx_core::f_is_endswith(
                vx_data_file::f_pathcurrent_from_os(),
                vx_core::t_string->vx_new_from_string("\\java")
              ),
              context
            )
        )
      )
    );
    return output;
  }

  vx_test::Type_testcase* f_string_read_from_file(vx_core::Type_context* context) {
    vx_test::Type_testcase* output = vx_test::t_testcase->vx_new(
      vx_core::t_string->vx_new_from_string(":passfail"), vx_core::c_false,
      vx_core::t_string->vx_new_from_string(":testpkg"), vx_core::t_string->vx_new_from_string("vx/data/file"),
      vx_core::t_string->vx_new_from_string(":casename"), vx_core::t_string->vx_new_from_string("string-read<-file"),
      vx_core::t_string->vx_new_from_string(":describelist"),
      vx_test::t_testdescribelist->vx_new(
        vx_test::t_testdescribe->vx_new(
          vx_core::t_string->vx_new_from_string(":describename"), vx_core::t_string->vx_new_from_string("(test\n \"testdata\"\n (string-read<-file\n  (file\n   :name \"string_read_from_file.txt\"\n   :path \"src/test/resources/vx\")))"),
          vx_core::t_string->vx_new_from_string(":testresult"),
            vx_test::f_test(
              vx_core::t_string->vx_new_from_string("testdata"),
              vx_data_file::f_string_read_from_file(
                vx_core::f_new(
                  vx_data_file::t_file,
                  vx_core::t_anylist->vx_new(
                    vx_core::t_string->vx_new_from_string(":name"),
                    vx_core::t_string->vx_new_from_string("string_read_from_file.txt"),
                    vx_core::t_string->vx_new_from_string(":path"),
                    vx_core::t_string->vx_new_from_string("src/test/resources/vx")
                  )
                ),
                context
              ),
              context
            )
        )
      )
    );
    return output;
  }

  vx_test::Type_testcaselist* test_cases(vx_core::Type_context* context) {
    std::vector<vx_core::Type_any*>* arraylisttestcase;
    arraylisttestcase->push_back(vx_data_file_test::f_boolean_exists_from_file(context));
    arraylisttestcase->push_back(vx_data_file_test::f_boolean_write_from_file_string(context));
    arraylisttestcase->push_back(vx_data_file_test::f_file_read_from_file(context));
    arraylisttestcase->push_back(vx_data_file_test::f_pathcurrent_from_os(context));
    arraylisttestcase->push_back(vx_data_file_test::f_string_read_from_file(context));
    vx_test::Type_testcaselist* output = vx_test::t_testcaselist->vx_new(arraylisttestcase);
    return output;
  }

  vx_test::Type_testcoveragesummary* test_coveragesummary() {
    return vx_test::t_testcoveragesummary->vx_new(
      vx_core::t_string->vx_new_from_string(":testpkg"), vx_core::t_string->vx_new_from_string("vx/data/file"),
      vx_core::t_string->vx_new_from_string(":constnums"), vx_test::t_testcoveragenums->vx_new(vx_core::t_string->vx_new_from_string(":pct"), vx_core::t_int->vx_new_from_int(100), vx_core::t_string->vx_new_from_string(":tests"), vx_core::t_int->vx_new_from_int(0), vx_core::t_string->vx_new_from_string(":total"), vx_core::t_int->vx_new_from_int(0)),
      vx_core::t_string->vx_new_from_string(":docnums"), vx_test::t_testcoveragenums->vx_new(vx_core::t_string->vx_new_from_string(":pct"), vx_core::t_int->vx_new_from_int(90), vx_core::t_string->vx_new_from_string(":tests"), vx_core::t_int->vx_new_from_int(10), vx_core::t_string->vx_new_from_string(":total"), vx_core::t_int->vx_new_from_int(11)),
      vx_core::t_string->vx_new_from_string(":funcnums"), vx_test::t_testcoveragenums->vx_new(vx_core::t_string->vx_new_from_string(":pct"), vx_core::t_int->vx_new_from_int(55), vx_core::t_string->vx_new_from_string(":tests"), vx_core::t_int->vx_new_from_int(5), vx_core::t_string->vx_new_from_string(":total"), vx_core::t_int->vx_new_from_int(9)),
      vx_core::t_string->vx_new_from_string(":ospacenums"), vx_test::t_testcoveragenums->vx_new(vx_core::t_string->vx_new_from_string(":pct"), vx_core::t_int->vx_new_from_int(0), vx_core::t_string->vx_new_from_string(":tests"), vx_core::t_int->vx_new_from_int(0), vx_core::t_string->vx_new_from_string(":total"), vx_core::t_int->vx_new_from_int(9)),
      vx_core::t_string->vx_new_from_string(":otimenums"), vx_test::t_testcoveragenums->vx_new(vx_core::t_string->vx_new_from_string(":pct"), vx_core::t_int->vx_new_from_int(0), vx_core::t_string->vx_new_from_string(":tests"), vx_core::t_int->vx_new_from_int(0), vx_core::t_string->vx_new_from_string(":total"), vx_core::t_int->vx_new_from_int(9)),
      vx_core::t_string->vx_new_from_string(":totalnums"), vx_test::t_testcoveragenums->vx_new(vx_core::t_string->vx_new_from_string(":pct"), vx_core::t_int->vx_new_from_int(45), vx_core::t_string->vx_new_from_string(":tests"), vx_core::t_int->vx_new_from_int(5), vx_core::t_string->vx_new_from_string(":total"), vx_core::t_int->vx_new_from_int(11)),
      vx_core::t_string->vx_new_from_string(":typenums"), vx_test::t_testcoveragenums->vx_new(vx_core::t_string->vx_new_from_string(":pct"), vx_core::t_int->vx_new_from_int(0), vx_core::t_string->vx_new_from_string(":tests"), vx_core::t_int->vx_new_from_int(0), vx_core::t_string->vx_new_from_string(":total"), vx_core::t_int->vx_new_from_int(2))
    );
  }

  vx_test::Type_testcoveragedetail* test_coveragedetail() {
    return vx_test::t_testcoveragedetail->vx_new(
      vx_core::t_string->vx_new_from_string(":testpkg"), vx_core::t_string->vx_new_from_string("vx/data/file"),
      vx_core::t_string->vx_new_from_string(":typemap"), vx_core::t_intmap->vx_new(
        vx_core::t_string->vx_new_from_string(":file"), vx_core::t_int->vx_new_from_int(0),
        vx_core::t_string->vx_new_from_string(":fileformat"), vx_core::t_int->vx_new_from_int(0)
      ),
      vx_core::t_string->vx_new_from_string(":constmap"), vx_core::t_intmap->vx_new(
  
      ),
      vx_core::t_string->vx_new_from_string(":funcmap"), vx_core::t_intmap->vx_new(
        vx_core::t_string->vx_new_from_string(":boolean-exists<-file"), vx_core::t_int->vx_new_from_int(1),
        vx_core::t_string->vx_new_from_string(":boolean-write<-file-any"), vx_core::t_int->vx_new_from_int(0),
        vx_core::t_string->vx_new_from_string(":boolean-write<-file-string"), vx_core::t_int->vx_new_from_int(1),
        vx_core::t_string->vx_new_from_string(":file-read<-file"), vx_core::t_int->vx_new_from_int(1),
        vx_core::t_string->vx_new_from_string(":name<-file"), vx_core::t_int->vx_new_from_int(0),
        vx_core::t_string->vx_new_from_string(":path<-file"), vx_core::t_int->vx_new_from_int(0),
        vx_core::t_string->vx_new_from_string(":pathcurrent<-os"), vx_core::t_int->vx_new_from_int(1),
        vx_core::t_string->vx_new_from_string(":pathfull<-file"), vx_core::t_int->vx_new_from_int(0),
        vx_core::t_string->vx_new_from_string(":string-read<-file"), vx_core::t_int->vx_new_from_int(1)
      )
    );
  }

  vx_test::Type_testpackage* test_package(vx_core::Type_context* context) {
    vx_test::Type_testcaselist* testcaselist = test_cases(context);
    vx_test::Type_testpackage* output = vx_test::t_testpackage->vx_new(
      vx_core::t_string->vx_new_from_string(":testpkg"), vx_core::t_string->vx_new_from_string("vx/data/file"), 
      vx_core::t_string->vx_new_from_string(":caselist"), testcaselist,
      vx_core::t_string->vx_new_from_string(":coveragesummary"), test_coveragesummary(),
      vx_core::t_string->vx_new_from_string(":coveragedetail"), test_coveragedetail()
    );
    return output;
  }

}
